soc/riscv/riscv-privilege/common/soc_common_irq.c

```diff
@@ -11,12 +11,30 @@
  */
 #include <irq.h>
 
+/**
+ * @brief Get an IRQ's level
```

Comment:
I think it'd be nice to have generic helpers in an `<interrupt_controller/multilevel.h>` or something in case we need to change the encoding later, but it's not a big deal.

(The current encoding can be pretty wasteful of space if there are IRQ chips with widely varying numbers of IRQs, so I don't expect it to last forever.)

Reply:
Yeah, felt like that would be a post cleanup after this goes in.